src/main/java/org/junit/runner/OrderWith.java

```diff
@@ -0,0 +1,26 @@
+package org.junit.runner;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.runner.manipulation.Ordering;
+
+/**
+ * When a test class is annotated with <code>&#064;OrderWith</code> or extends a class annotated
+ * with <code>&#064;OrderWith</code>, JUnit will order the tests in the test class (and child
+ * test classes, if any) using the ordering defined by the {@link Ordering} class.
+ *
+ * @since 4.13
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface OrderWith {
```

Comment:
A couple musings here (which I may not finish all at once)

1) I'd be strongly tempted to make this an annotated static field instead/in-addition.  Why?  Well, consider the case (which I've written several times in the past) where you try to run the tests from the fastest to the slowest, based on a record of previous runtimes kept in a file.  It probably makes sense to allow the location of the file to be specified by the test writer.  I think that with the current pull request, this would be 100% impossible.  One COULD imagine a different spec of this annotation where what's returned is not an Ordering, but a functional interface that takes a Description and returns an Ordering, so that one could write a Fastest Ordering that looks for a @TestTimesFile annotation on the Description to know what file to use, BUT...

It might be so much easier to instead just say:

@OrderWith public static Ordering ordering = new FastestOrdering("/tmp/test-times.txt");

or some such.